flutter/lib/src/event_processor/platform_exception_event_processor.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:sentry/sentry.dart';
+
+/// Add code & message from [PlatformException] to [SentryException]
+class PlatformExceptionEventProcessor implements EventProcessor {
+  @override
+  FutureOr<SentryEvent?> apply(SentryEvent event, {Hint? hint}) {
+    final exceptions = <SentryException>[];
+
+    for (SentryException exception in (event.exceptions ?? [])) {
+      final platformException = exception.throwable;
+      if (platformException is PlatformException) {
+        exception = _enrich(exception, platformException);
+      }
+      exceptions.add(exception);
+    }
+
+    return event.copyWith(exceptions: exceptions);
+  }
+
+  SentryException _enrich(
+      SentryException sentryException, PlatformException platformException) {
+    final data = Map<String, dynamic>.from(
+      sentryException.mechanism?.data ?? {},
+    );
+    data['platformExceptionCode'] = platformException.code;
+    if (platformException.message != null) {
+      data['platformExceptionMessage'] = platformException.message;
```

Comment:
```suggestion
    data['code'] = platformException.code;
    if (platformException.message != null) {
      data['message'] = platformException.message;
```

We can be more generic here, the `mechanism` has a type already.
[PHP](https://github.com/getsentry/sentry-php/pull/1451) also uses `code`.
If there's already a mechanism, the type of the exception will be `PlatformException` anyway.